packages/@aws-cdk/aws-codepipeline-actions/test/codecommit/test.codecommit-source-action.ts

```diff
@@ -224,6 +224,51 @@ export = {
 
       test.done();
     },
+
+    'allows using a Token for the branch name'(test: Test) {
+      const stack = new Stack();
+
+      const branch = Lazy.stringValue({ produce: () => 'branch' });
+
+      const p = new codepipeline.Pipeline(stack, 'P');
+
+      const r = new codecommit.Repository(stack, 'R', {
+        repositoryName: 'repository',
+      });
+
+      const artifact = new codepipeline.Artifact();
+      const ccSourceAction = new cpactions.CodeCommitSourceAction({
+        actionName: 'CodeCommit',
+        repository: r,
+        branch: branch,
+        output: artifact,
+      });
+
+      p.addStage({
+        stageName: 'Source',
+        actions: [ccSourceAction],
+      });
+      p.addStage({
+        stageName: 'Build',
+        actions: [
+          new cpactions.CodeBuildAction({
+            actionName: 'Build',
+            project: new codebuild.PipelineProject(stack, 'CodeBuild'),
+            input: artifact,
+          }),
+        ],
+      });
```

Comment:
Gotta say, not a hue fan of these one-letter variable names 🙂. Do we actually need them for anything? Can't we do the setup in one expression?

```suggestion
      const sourceOutput = new codepipeline.Artifact();
      new codepipeline.Pipeline(stack, 'P', {
        stages: [
          {
            stageName: 'Source',
            actions: [
              new cpactions.CodeCommitSourceAction({
                actionName: 'CodeCommit',
                repository: new codecommit.Repository(stack, 'R', {
                  repositoryName: 'repository',
                }),
                branch: Lazy.stringValue({ produce: () => 'my-branch' }),
                output: sourceOutput,
              }),
            ],
          },
          {
            stageName: 'Build',
            actions: [
              new cpactions.CodeBuildAction({
                actionName: 'Build',
                project: new codebuild.PipelineProject(stack, 'CodeBuild'),
                input: sourceOutput,
              }),
            ],
          },
        ],
      });
```